docs/usage/self-hosted-configuration.md

```diff
@@ -59,6 +59,39 @@ But before you disable templating completely, try the `allowedPostUpgradeCommand
 
 ## allowScripts
 
+## allowedHeaders
+
+List of headers that are allowed to be forwarded to the HTTP request. This is useful when the `registryUrlTemplate` employs a specific authentication system. By default, all headers starting with "X-" are allowed, but you can include additional headers using this option. If declared, it will override the default "X-" allowed headers. `allowedHeaders` is an array of RegExp represented as strings due to JSON serialization
```

Comment:
```suggestion
List of headers that are allowed to be forwarded to the HTTP request. This is useful when the `registryUrlTemplate` employs a specific authentication system. By default, all headers starting with "X-" are allowed, but you can include additional headers using this option. If declared, it will override the default "X-" allowed headers. `allowedHeaders` is an array of minimatch-compatible globs or re2-compatible regex represented as strings.

Examples:
- `/X/` - any header with `abc` in the name (regex)
- `!/X/` - any header without `abcd` in the name (regex)
- `X-*` - any header starting with `X-` (glob pattern)
- `X` - only the header matching exactly `X` (exact match glob)
```